### PR TITLE
feat(#36): add auth middleware

### LINKED ISSUE #36: Auth Header
### Is your feature request related to a problem? Please describe:
As we are not yet concerned about auth, we only want a binary check for secured routes, so it's easier for the frontend to integrate.

### Describe the solution you'd like
Add a middleware for, that checks if the `Authorization` header is set and returns a 400 when not.

### Additional Comments
The Create and Update routes for ERs are a great point to start using it.

### PR DESCRIPTION
**Pull request by FoseFx**
Closes #36

### CODE REVIEW

File: routes/rooms.go [hunk 1]
```diff
@@ -47,6 +48,7 @@ func CreateEmergencyRoom(ctx *gin.Context) {
 // @Summary     update an emergency room by id
 // @Tags 		emergency-rooms
 // @Produce 	json
+// @Param       authorization header    string              true    "Bearer: <TOKEN>"
```
Comment (use-to) [inline routes/rooms.go:51]: This looks different to line 38. Maybe fix formatting here!

**Review by use-to — CHANGES_REQUESTED** [review]

File: routes/rooms.go [hunk 2]
```diff
@@ -93,28 +94,28 @@ func CreateEmergencyRoom(ctx *gin.Context) {
...
-// @Produce 	 json
-// @Param        authorization  header  string      true    "Bearer: <TOKEN>"
-// @Param        id   path 		string  			true	"Emergency Room's ID"
-// @Success      200  {object} 	StatusResponse
-// @Failure      501  {object}  HTTPErrorResponse
-// @Router       /er/{id} [delete]
+// @Summary    delete an emergency room by id
+// @Tags       emergency-rooms
+// @Produce    json
+// @Param      authorization              header       string                true    "Bearer: <TOKEN>"
```
Comment (use-to) [inline routes/rooms.go:114]: Missing spaces here. Look at line 100 for example. Be consistent here, please!

**Review by use-to — CHANGES_REQUESTED** [review]

File: routes/rooms.go (hunk 2 above)
Comment (FoseFx) [reply routes/rooms.go:114]: spacing seems fine for me? (both on gh and in IDE)

**Review by use-to — APPROVED** [review]
Look at #44 
We have to fix those comments/definitions! As mentioned above... 😡

**Comment by use-to** [comment]
image

**Pull request merged**